### DISCUSSION TITLE
API and Webhooks: October Community Check-in 👻

### DISCUSSION THREAD

**Discussion by ghostinhershell** [community/community#34998] [Apps, API and Webhooks]
Hey guys, checking in for our API and Webhooks mini check-in ✅

### 🧠​​ Pros in the Community
Big kudos to these community contributors 👏
- @LangLangBart for showing us [how to retrieve GitHub edit history via the API](https://github.com/orgs/community/discussions/33551#discussioncomment-3689510) 📚
- Shoutout to @jdevstatic and @airtower-luna for coming through and helping a user figure out [where a strange URL originated from](https://github.com/orgs/community/discussions/32478) 🕵️‍♂️

### 🗞️What’s New?
Have you heard? Custom Repository Role Creation APIs are available in public beta 🗣️
- You can read more in our announcement [here](https://github.blog/changelog/2022-09-07-custom-repository-role-creation-apis-are-now-available-in-public-beta/).
- TLDR: REST API endpoints to create and update [custom repository roles](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/managing-custom-repository-roles-for-an-organization) are available in a public beta for GitHub Enterprise Cloud customers. These new endpoints build on the existing custom repository role APIs that allow assignment of those roles to a [team](https://docs.github.com/en/enterprise-cloud@latest/rest/teams/teams#add-or-update-team-repository-permissions) or [user](https://docs.github.com/en/enterprise-cloud@latest/rest/collaborators/collaborators#add-a-repository-collaborator).

### 💡Inspiration Station
Fresh content to give you new ideas! 
- Public APIs: Ever needed to find a free API for your next project, website, or app? Check out the [Public APIs Repository](https://github.com/public-apis-dev/public-apis) 📃
- New to open source? [Here’s how to find projects that need help and start making impactful contributions](https://github.com/collections/choosing-projects) 🏗️

### 🧹Housekeeping
Help us make the most out of Community on Discussions, take our survey!
- Tell us what you think of github.com/community in a [brief survey](https://github.com/community/community/discussions/33536) **before October 4th** so we can continue to improve your community experience 👷‍♂️

---

Have any projects you built that you would like to share? Link them below! Cheers 😄

**Comment by 3kh0** [comment]
When can we use repo roles for personal accounts?

**Reply by RONIE70** [reply]
Like personal accounts, organizations can own resources such as repositories, packages, and projects. However, you cannot sign into an organization. Instead, each person signs into their own personal account, and any actions the person takes on organization resources are attributed to their personal account. Each personal account can be a member of multiple organizations.

The personal accounts within an organization can be given different roles in the organization, which grant different levels of access to the organization and its data. All members can collaborate with each other in repositories and projects, but only organization owners and security managers can manage the settings for the organization and control access to the organization's data with sophisticated security and administrative features. For more information, see "[Roles in an organization](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization)" and "[Keeping your organization secure](https://docs.github.com/en/organizations/keeping-your-organization-secure)."

**Reply by 3kh0** [reply]
@RONIE70 I know, but you can do this say, on the repo level on your personal account, you can only do it on a organization.

**Comment by ghost** [comment]
[]()

**Comment by ghostinhershell** [comment]
Hey Community 👋 

Missed out on our last community check-in? Check out our first community check-in here: [A Big Hi, Hello 👋: API & Webhooks Community Check-In](https://github.com/orgs/community/discussions/31405)
